modules/reitit-core/src/reitit/impl.cljc

```diff
@@ -9,6 +9,51 @@
      (:import (java.net URLEncoder URLDecoder)
               (java.util HashMap Map))))
 
+;;
+;; path-update
+;;
+
+(defn -match [path path-map]
+  (letfn [(match [x f] (if (fn? f) (f x) (= x f)))]
+    (reduce
+     (fn [_ [ps f]]
+       (let [match (loop [[p & pr] path, [pp & ppr] ps]
+                     (cond (and p pp (match p pp)) (recur pr ppr)
+                           (= nil p pp) true))]
+         (when match (reduced f))))
+     nil path-map)))
+
+(defn -path-vals [m path-map]
+  (letfn [(-path-vals [l p m]
+            (reduce
+             (fn [l [k v]]
+               (let [p' (conj p k)
+                     f (-match p' path-map)]
+                 (cond
+                   f (cons [p' (f v)] l)
+                   (and (map? v) (seq v)) (-path-vals l p' v)
+                   :else (cons [p' v] l))))
+             l m))]
+    (reverse (-path-vals [] [] m))))
```

Comment:
you can skip the  if you `conj` into a vector of `cons`ing

Reply:
good point! original code [here](https://github.com/metosin/potpuri/blob/master/src/potpuri/core.cljc#L47).